Final Project/Assets/Geometry/RTTriangle.cs

```diff
@@ -48,43 +44,27 @@ public RTTriangle_t GetGeometry()
         };
     }
 
-
-    private bool IsAnyChangesOnVertices()
-    {
-        var _hasChanged = false;
-
-        if (m_vertices0 != m_cachedVert0)
-        {
-            _hasChanged = true;
-            m_cachedVert0 = m_vertices0;
-        }
-        
-        if (m_vertices1 != m_cachedVert1)
-        {
-            _hasChanged = true;
-            m_cachedVert1 = m_vertices1;
-        }
-        
-        if (m_vertices2 != m_cachedVert2)
-        {
-            _hasChanged = true;
-            m_cachedVert2 = m_vertices2;
-        }
-
-        return _hasChanged;
-    }
-
     
     private void UpdateTriangle()
     {
-        m_worldVert0 = m_vertices0 + transform.position;
-        m_worldVert1 = m_vertices1 + transform.position;
-        m_worldVert2 = m_vertices2 + transform.position;
+        m_worldVert0 = (transform.localRotation * m_vertices0) + transform.localPosition;
+        m_worldVert1 = (transform.localRotation * m_vertices1) + transform.localPosition;
+        m_worldVert2 = (transform.localRotation * m_vertices2) + transform.localPosition;
+
+        // m_worldVert0 = transform.localToWorldMatrix * m_vertices0;
```

Comment:
Hm ... this does not make sense to me ... 

    localToWorldMatrix = localScale * localRotation * localTranslation

What you are doing is (should be) precisely what localToWorldMatrix is doing. Let me take a closer look this evening.